### DISCUSSION TITLE
Closest words from embedding vector

### DISCUSSION THREAD

**Discussion by rhohndorf** [ggml-org/llama.cpp#1942] [General]
Hi,
is it possible to get the n closest words for an embedding vector ?

**Comment by Green-Sky** [comment]
1. yes, you can build and use an embedding database for that (see [usearch](https://github.com/unum-cloud/usearch) [hnswlib](https://github.com/nmslib/hnswlib) )
2. as outlined in #899 , gpt based models have exceptionally bad performance when used as sentence embedder. I recommend you use something else, like https://github.com/skeskinen/bert.cpp for sbert embeddings, or https://github.com/monatis/clip.cpp for text and image embeddings.
